api/staged_products_property_collection_helpers.go

```diff
@@ -0,0 +1,142 @@
+package api
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+)
+
+type updatedPropertyCollectionItem struct {
+	Data map[string]interface{}
+}
+type updatedPropertyCollection []updatedPropertyCollectionItem
+
+func (item updatedPropertyCollectionItem) getFieldValue(fieldName string) string {
+	if value, ok := item.Data[fieldName].(string); ok {
+		return value
+	}
+	return ""
+}
+
+func (item updatedPropertyCollectionItem) setFieldValue(fieldName string, value string) {
+	item.Data[fieldName] = value
+}
+
+func parseUpdatedPropertyCollection(updatedProperty interface{}) (updatedPropertyCollection, error) {
```

Comment:
In general, we try to avoid nested if statements in go. This helps with readability and maintainability as well, though it takes more lines of code.

Alternatively, this function could be written like
```
func parseUpdatedPropertyCollection(updatedProperty interface{}) (updatedPropertyCollection, error) {
	var collection updatedPropertyCollection

	updatedPropertyAsMap, ok := updatedProperty.(map[string]interface{})
	if !ok {
		return nil, fmt.Errorf("parseUpdatedPropertyCollection: failed to convert %v to map[string]interface{}", updatedProperty)
	}

	rawItems := updatedPropertyAsMap["value"]

	rawItemSlice, ok := rawItems.([]interface{})
	if !ok {
		return nil, fmt.Errorf("parseUpdatedPropertyCollection: failed to convert %v to []interface{}", rawItems)
	}

	for _, item := range rawItemSlice {
		itemMap, ok := item.(map[string]interface{})
		if !ok {
			return nil, fmt.Errorf("parseUpdatedPropertyCollection: failed to convert %v to map[string]interface{}", item)
		}
		
		collection = append(collection, updatedPropertyCollectionItem{Data: itemMap})
	}
	
	return collection, nil
}
```

Reply:
Done.

Makes for an interesting inversion of the `try { happyPath() } catch exception {}` flow I'm used to when reading the code; but removing the nesting is a big plus